lychee-lib/src/utils/request.rs

```diff
@@ -48,8 +49,20 @@ fn create_request(
 ///   to create a valid URI.
 /// - If a URI cannot be created from the file path.
 /// - If the source is not a file path (i.e. the URI type is not supported).
-fn try_parse_into_uri(raw_uri: &RawUri, source: &InputSource, base: &Option<Base>) -> Result<Uri> {
-    let text = raw_uri.text.clone();
+fn try_parse_into_uri(
+    raw_uri: &RawUri,
+    source: &InputSource,
+    root_path: &Option<PathBuf>,
+    base: &Option<Base>,
+) -> Result<Uri> {
+    let mut text = raw_uri.text.clone();
+    if text.starts_with('/') {
+        if let Some(path) = root_path {
+            if let Some(path_str) = path.to_str() {
+                text = format!("{path_str}{text}");
+            }
+        }
+    }
```

Comment:
Maybe we could put this logic into a separate function/method and add some tests for it. Not sure about the function name yet &mdash; `make_absolute`?

Reply:
makes sense, I added the function and tests, you probably won't like the name I gave it (sorry, I'm a verbose Java dev 😂), I'm happy to rename it, just let me know what you'd like

Reply:
oh, I like explicit names; no worries